apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -912,14 +928,22 @@ block_txs_count_pending(_Config) ->
                     get_block_txs_count_preset("pending"),
     rpc(application, set_env, [aecore, expected_mine_rate,
                                60 * 60 * 1000]), % aim at one block an hour
+    InsertedTxsCount = add_spend_txs(),
     rpc(aec_conductor, start_mining, []),
-    rpc(aec_conductor, start_mining, []),
-    timer:sleep(100),% so the miner is started
-    add_spend_txs(),
-    {ok, PendingBlock} = get_pending_block(),
-    TxsCount = length(aec_blocks:txs(PendingBlock)), 
-    {ok, 200, #{<<"count">> := TxsCount}} =
-        get_block_txs_count_preset("pending"),
+    GetPending =
+        fun TryToGet(Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Attempts) ->
+                case get_block_txs_count_preset("pending") of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} ->
+                        timer:sleep(100),
+                        TryToGet(Attempts -1)
+                end
+        end,
+    ct:log("Inserted transactions count ~p", [InsertedTxsCount]),
+    {ok, #{<<"count">> := TxsCount}} = GetPending(10),
```

Comment:
I suggest adding comment on idea behind test design like "`The custom set high expected inter-block interval (`expected_mine_rate`) makes almost certain that the retrieved pending block is the first after having added the spend txs."